internal/agent/device/hook/default_hooks.go

```diff
@@ -52,7 +52,12 @@ func defaultBeforeUpdateHooks() []v1alpha1.DeviceUpdateHookSpec {
 			Description: util.StrToPtr("Bring down a multi-container system based on the provided YAML podman-compose definition file"),
 			OnFile:      &[]v1alpha1.FileOperation{v1alpha1.FileOperationUpdate, v1alpha1.FileOperationRemove},
 			Actions: []v1alpha1.HookAction{
-				marshalExecutable("podman-compose -f {{ .FilePath }} down", nil,
+				marshalExecutable(`output=$(podman-compose -f {{ .FilePath }} down 2>&1); exit_code=$? ; \
+										if [[ $exit_code -eq 0 ]] || ( [[ ! -f {{ .FilePath }} ]] && \
+                                            ! ( podman ps -a --format=json | jq -r '.[] | .Labels."com.docker.compose.project.config_files"' | grep -q  -E '^{{ .FilePath }}$' )) ; then  \ 
+											exit 0; \
```

Comment:
Is it possible to create a podman compose client and push this type of logic down? Can we track that and unit tests as a follow-up?

Reply:
In general, using hooks for launching processes/applications since it makes lifecycle very hard to maintain.
In this case if we try apply desired spec twice for some reason, the second time will fail.
I think we should consider tools that provide better way to manage the application lifecycle (example systemd services), or provide better way to manage/monitor applications.  Hooks do not provide this functionality.

Reply:
Right there is a full application level status defined in DDD which is TODO. This future application manager will handle application-level lifecycle status. I would consider this hooks implementation for applications an interim state. In the future, this application manager will trigger some lifecycle actions off hooks observations and handle lifecycles directly. But today could we make this better, even just with unit tests?

Reply:
I'll try adding tests